bpfd-operator/controllers/bpfd-agent/bpfd-agent.go

```diff
@@ -241,147 +243,142 @@ func (r *BpfProgramReconciler) reconcileBpfProgramConfig(ctx context.Context,
 
 	isNodeSelected = selector.Matches(nodeLabelSet)
 
-	programKey := internal.ProgramKey{
-		Name:        BpfProgramConfig.Spec.Name,
-		ProgType:    BpfProgramConfig.Spec.Type,
-		AttachPoint: internal.StringifyAttachType(&BpfProgramConfig.Spec.AttachPoint),
-	}
+	// Convert BpfProgramConfig object to a list of BpfProgramConfig, which maps any non-interface
+	// (like pod label or primary interface flag) into its actual interface or list of interfaces.
+	bpfProgramConfigList := internal.ConvertToBpfProgramConfigList(BpfProgramConfig)
```

Comment:
OK. After discussion, @astoycos and @Billy99 both agree, but will done in a follow-up PR because it is a lot of re-work. The follow-up PR, which will add support for specifying a pod label will include a new internal type that both the BpfProgramConfig Spec and bpfd list result structs map too.